timeless/reservations/views.py

```diff
@@ -116,37 +117,43 @@ def list():
     return render_template("restaurants/tables/list.html")
 
 
-@BP.route("/create", methods=("GET", "POST"))
-def create():
-    """ Create new reservation """
-    """
-    @todo #380:30min Continue. When we use the form.validate() function the
-     status (enum: timeless/restaurants/models.py) is displaying the
-     following error: 'status': ['Invalid Choice: could not coerce', 'Not a
-     valid choice']. Removes the commented if to see the error. Also refactor
-     the function using genericViews, as explained in puzzles #134 and #137.
-    """
-    form = ReservationForm(request.form)
-    try:
-        """if request.method == "POST" and form.validate():"""
-        if request.method == "POST":
-            reservation = Reservation(
-                start_time=request.form["start_time"],
-                end_time=request.form["end_time"],
-                num_of_persons=request.form["num_of_persons"],
-                comment=request.form["comment"],
-                status=request.form["status"]
-            )
-            DB.session.add(reservation)
-            DB.session.commit()
-            return redirect(url_for("reservations.list"))
-        else:
-            flash("Error: ", form.errors)
-    except Exception as error:
-        flash("Error: ", error)
-
-    return render_template("reservations/create_edit.html", error=form.errors,
-                            action="create", form=form)
+class CreateReservation(views.CrudAPIView):
```

Comment:
@emilianodellacasa why ``CrudAPIView`` and not ``CreateView, like employees are created, [here](https://github.com/timelesslounge/timelessis/blob/master/timeless/employees/views.py#L17)? I'm a bit confused :D

Reply:
@amihaiemil Reservations use a different pattern, as they will be almos exclusively used in React pages, this is the reason why `CrudAPIView` is used